libcontainer/apparmor/apparmor.go

```diff
@@ -24,15 +21,29 @@ func IsEnabled() bool {
 	return false
 }
 
+func setprocattr(tid int, attr, value string) error {
+	path := fmt.Sprintf("/proc/%d/attr/%s", tid, attr)
+
+	f, err := os.OpenFile(path, os.O_WRONLY, 0)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	_, err = fmt.Fprintf(f, "%s", value)
+	return err
+}
+
 // ApplyProfile will apply the profile with the specified name to the process after
 // the next exec.
 func ApplyProfile(name string) error {
 	if name == "" {
 		return nil
 	}
-	cName := C.CString(name)
-	defer C.free(unsafe.Pointer(cName))
-	if _, err := C.aa_change_onexec(cName); err != nil {
+
+	// mimick aa_change_onexec from libapparmor in Go
+	value := "exec " + name
+	if err := setprocattr(unix.Gettid(), "exec", value); err != nil {
```

Comment:
This looks /okay/ overall, but I would prefer if we had a function called `ChangeOnExec` (or whatever) which wraps these two lines -- to make it clearer what we are mimicing.

Reply:
Agree. Does it need to be exported or could it also be `changeOnExec`?

Reply:
PTAL

Reply:
Yeah, there's no need to export it I guess.